### PR TITLE
Create flutter specific leak troubleshooting guidance.

### PR DESCRIPTION
**Pull request by polina-c**

### CODE REVIEW

**Comment by polina-c** [comment]
There is no tests, because this PR updates testing code and documentation only.

File: docs/contributing/testing/Leak-tracking.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+It is ok to opt out a test from leak tracking in these cases:
+
+1. Hacky tests
+
+    A test leaks because it uses APIs in non-recommended way
+    and there is no simple way to either test what is needed
+    in a recommended way or to dispose the objects.
+
+2. Thrown exceptions
```
Comment (Piinks) [inline docs/contributing/testing/Leak-tracking.md:24]: This is a common test case, is there guidance on how to handle these without opting out of leak tracking?

File: docs/contributing/testing/Leak-tracking.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    Some leaks happen because exceptions are thrown and
+    the code did not finalize properly.
+
+    While some exceptions should be finalized properly
+    and should not result in leaking objects,
+    the project to make sure disposables are disposed
+    in exceptional code paths was not prioritized yet.
```
Comment (Piinks) [inline docs/contributing/testing/Leak-tracking.md:32]: Is there an issue tracking this that folks can refer to for updates?

File: docs/contributing/testing/Leak-tracking.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+Leak tracking is disabled for local test runs and for all bots except two:
+
+- [Windows framework_tests_libraries_leak_tracking](https://github.com/flutter/flutter/blob/9441f8f6c806fb0a3b7d058a40b5e59c373e6055/.ci.yaml#L5553)
+- [Windows framework_tests_widgets_leak_tracking](https://github.com/flutter/flutter/blob/9441f8f6c806fb0a3b7d058a40b5e59c373e6055/.ci.yaml#L5640C11-L5640C56)
+
+To enable leak tracking for flutter tests, pass
+`--dart-define LEAK_TRACKING=true` to `flutter test`.
+
+You can see the bot's status on [Flutter dashboard](https://flutter-dashboard.appspot.com/#/build).
```
Comment (Piinks) [inline docs/contributing/testing/Leak-tracking.md:51]:
```suggestion
You can see the bot's status on the [Flutter build dashboard](https://flutter-dashboard.appspot.com/#/build).
```

File: docs/contributing/testing/Leak-tracking.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+## Where leak tracking is configured?
+
+Leak tracker is configured in [flutter_test_config.dart](https://github.com/flutter/flutter/blob/9441f8f6c806fb0a3b7d058a40b5e59c373e6055/packages/flutter/test/flutter_test_config.dart#L45).
+
+See [leak_tracker documentation](https://github.com/dart-lang/leak_tracker/blob/main/doc/leak_tracking/TROUBLESHOOT.md)
+on how to enable/disable leak tracking for individual tests and files.
+
+## What are defaults?
+
+Leak tracking is disabled for local test runs and for all bots except two:
```
Comment (Piinks) [inline docs/contributing/testing/Leak-tracking.md:43]:
```suggestion
Leak tracking is enabled on two test shards in Flutter's testing infrastructure:
```

File: docs/contributing/testing/Leak-tracking.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+- [Windows framework_tests_libraries_leak_tracking](https://github.com/flutter/flutter/blob/9441f8f6c806fb0a3b7d058a40b5e59c373e6055/.ci.yaml#L5553)
+- [Windows framework_tests_widgets_leak_tracking](https://github.com/flutter/flutter/blob/9441f8f6c806fb0a3b7d058a40b5e59c373e6055/.ci.yaml#L5640C11-L5640C56)
+
+To enable leak tracking for flutter tests, pass
```
Comment (Piinks) [inline docs/contributing/testing/Leak-tracking.md:48]: Is this for local testing?

**Review by Piinks — COMMENTED** [review]
Also would be good to include an example of the failure output from a leak tracking test so folks can identify it.

File: packages/flutter/test/flutter_test_config.dart [hunk 6]
```diff
@@ -44,6 +44,8 @@ Future<void> testExecutable(FutureOr<void> Function() testMain) {
   if (_isLeakTrackingEnabled()) {
     LeakTesting.enable();
     LeakTracking.warnForUnsupportedPlatforms = false;
+    LeakTracking.troubleshootingDocumentationLink =
```
[the remark this answers is not in the document]
Comment (polina-c) [reply packages/flutter/test/flutter_test_config.dart:47]: Added comment.

File: docs/contributing/testing/Leak-tracking.md (hunk 1 above)
Comment (polina-c) [reply docs/contributing/testing/Leak-tracking.md:24]: not yet

File: docs/contributing/testing/Leak-tracking.md (hunk 2 above)
Comment (polina-c) [reply docs/contributing/testing/Leak-tracking.md:32]: Created: https://github.com/flutter/flutter/issues/157470

File: docs/contributing/testing/Leak-tracking.md (hunk 5 above)
Comment (polina-c) [reply docs/contributing/testing/Leak-tracking.md:48]: for both local testing and bots

**Comment by stuartmorgan** [comment]
> There is no tests, because this PR updates testing code and documentation only.

Isn't this changing user-facing behavior of a user-facing package that is intended for use in tests, rather than code that is just part of one of our own tests? If it's the former, that's not considered a test for exemption purposes.

**Comment by polina-c** [comment]
> > There is no tests, because this PR updates testing code and documentation only.
> 
> Isn't this changing user-facing behavior of a user-facing package that is intended for use in tests, rather than code that is just part of one of our own tests? If it's the former, that's not considered a test for exemption purposes.

The modified code is under 'test'.
It changes information that developers of Flutter Framework will see when they run unit tests of Flutter Framework.
All other people will not get any change.
All other code of flutter_test_config.dart is not test covered.

Does it help?

**Comment by stuartmorgan** [comment]
test-exempt: is a test

(Sorry about that, my brain got confused by the `flutter_test` in the filename, and read the path as `packages/flutter_test/...` rather than `packages/flutter/test/...`. Which I should have realized was wrong since it didn't have a `/lib/`.)

**Review by Piinks — APPROVED** [review]
LGTM

**Pull request merged**